tests/JustSaying.Extensions.DependencyInjection.AwsCore.Tests/ClientFactoryTests.cs

```diff
@@ -31,76 +41,88 @@ public void ShouldReadConfigFromFile()
-        Environment.SetEnvironmentVariable("AWS_SHARED_CREDENTIALS_FILE", null);
-        Environment.SetEnvironmentVariable("AWS_PROFILE", null);
 
     }
 
     [Fact]
-    public void ShouldReadFromSpecifiedProfile()
+    public void ShouldReadFromConfigValues()
     {
-        var bin = AppContext.BaseDirectory;
-        Environment.SetEnvironmentVariable("AWS_CONFIG_FILE", $"{bin}/config");
-        Environment.SetEnvironmentVariable("AWS_SHARED_CREDENTIALS_FILE", $"{bin}/credentials");
-        Environment.SetEnvironmentVariable("AWS_PROFILE", "local");
+        //Arrange
         var config = new ConfigurationBuilder()
-            .AddEnvironmentVariables()
+            .AddInMemoryCollection(new Dictionary<string, string>
+            {
+                {"AWS:Profile", "local"},
+                {"AWS:ProfilesLocation", _credentialsFile},
+                {"AWS:Region", "us-east-1"},
+                {"AWS:ServiceURL", "http://profile.test/"},
+                {"AWS:UseHttp", "true"}
+            })
             .Build();
         var serviceCollection = new ServiceCollection();
-        serviceCollection.AddJustSyaingWithAwsConfig(config, (_) => { });
+        serviceCollection.AddJustSayingWithAwsConfig(config, (_) => { });
+
+        //Act
         var provider = serviceCollection.BuildServiceProvider();
         var clientFactory = provider.GetRequiredService<IAwsClientFactory>();
         var snsClient = clientFactory.GetSnsClient(RegionEndpoint.USEast1);
-        Assert.True(snsClient.Config.ServiceURL.Equals("http://profile.test/") == true);
-    }
 
+        //Assert
+        Assert.True(snsClient.Config.RegionEndpoint == null);
+        Assert.True(snsClient.Config.ServiceURL.Equals("http://profile.test/") == true, snsClient.Config.ServiceURL);
+    }
 
     [Fact]
     public void CanSetCorrectRegion()
     {
-        var bin = AppContext.BaseDirectory;
-        Environment.SetEnvironmentVariable("AWS_CONFIG_FILE", $"{bin}/config");
-        Environment.SetEnvironmentVariable("AWS_SHARED_CREDENTIALS_FILE", $"{bin}/credentials");
-        Environment.SetEnvironmentVariable("AWS_PROFILE", "local");
+        //Arrange
         var config = new ConfigurationBuilder()
             .AddEnvironmentVariables()
             .Build();
         var serviceCollection = new ServiceCollection();
-        serviceCollection.AddJustSyaingWithAwsConfig(config, (_) => { });
+        serviceCollection.AddJustSayingWithAwsConfig(config, (_) => { });
+
+        //Act
         var provider = serviceCollection.BuildServiceProvider();
+        var clientFactory = provider.GetRequiredService<IAwsClientFactory>();
+        var sqsClient = clientFactory.GetSqsClient(RegionEndpoint.EUWest2);
+        var snsClient = clientFactory.GetSnsClient(RegionEndpoint.EUWest2);
 
+        //Assert
+        Assert.True(sqsClient.Config.RegionEndpoint == RegionEndpoint.EUWest2);
+        Assert.True(snsClient.Config.RegionEndpoint == RegionEndpoint.EUWest2);
```

Comment:
Same assertion twice - incorrect second assertion, or just duplicated?

Reply:
One is for the SNS client the other is for the SQS client. Will double check the assignments.

Reply:
Ah my bad, I just misread it.